### PR TITLE
Add MSSQL Host Assign SLA Function

### PR DESCRIPTION
**Pull request by Draper1**
# Description

Adding functionality to allow assigning SLA to a MSSQL Host and all subsequent instances under the host.

## Related Issue

Related Issue: https://github.com/rubrikinc/rubrik-sdk-for-python/issues/70

## Motivation and Context

Allows customers to apply SLA to an overall host

## How Has This Been Tested?

```
[2019-05-13 23:44:42,836] [DEBUG] -- Node IP: <ip_redacted>
[2019-05-13 23:44:42,836] [DEBUG] -- Username: <user_redacted>
[2019-05-13 23:44:42,836] [DEBUG] -- Password: *******

>>> 
>>> object_type = 'mssql_host'
>>> object_name = '<host_redacted>'
>>> sla_name = 'Gold'
>>> logBackupFrequencyInSeconds = 600
>>> logRetentionHours = 12
>>> copyOnly = False
>>> 
>>> assignsla = rubrik.assign_sla(object_name, sla_name, object_type, logBackupFrequencyInSeconds, logRetentionHours, copyOnly)
[2019-05-13 23:44:43,234] [DEBUG] -- assign_sla: Searching the Rubrik cluster for the SLA Domain 'Gold'.
[2019-05-13 23:44:43,234] [DEBUG] -- object_id: Getting the object id for the sla object 'Gold'.
[2019-05-13 23:44:43,234] [DEBUG] -- Creating the authorization header using the provided username and password.
[2019-05-13 23:44:43,235] [DEBUG] -- GET https://<ip_redacted>/api/v1/sla_domain?primary_cluster_id=local&name=Gold
[2019-05-13 23:44:45,330] [DEBUG] -- <Response [200]>

[2019-05-13 23:44:45,331] [DEBUG] -- Searching the Rubrik cluster for the current hosts.
[2019-05-13 23:44:45,331] [DEBUG] -- Creating the authorization header using the provided username and password.
[2019-05-13 23:44:45,331] [DEBUG] -- GET https://<ip_redacted>/api/v1/host?operating_system_type=Windows&primary_cluster_id=local
[2019-05-13 23:44:47,068] [DEBUG] -- <Response [200]>

[2019-05-13 23:44:47,069] [DEBUG] -- assign_sla: Searching the Rubrik cluster for the MSSQL Instance '<host_redacted>'.
[2019-05-13 23:44:47,069] [DEBUG] -- Creating the authorization header using the provided username and password.
[2019-05-13 23:44:47,069] [DEBUG] -- GET https://<ip_redacted>/api/v1/mssql/instance?root_id=Host:::32c0637b-6398-4f66-816c-044b2073b3cf
[2019-05-13 23:44:49,469] [DEBUG] -- <Response [200]>

[2019-05-13 23:44:49,470] [DEBUG] -- assign_sla: Determing the SLA Domain currently assigned to the MSSQL Instance 'MSSQLSERVER'.
[2019-05-13 23:44:49,470] [DEBUG] -- Creating the authorization header using the provided username and password.
[2019-05-13 23:44:49,470] [DEBUG] -- GET https://<ip_redacted>/api/v1/mssql/instance/MssqlInstance:::1b906a15-8197-4408-9ae3-109068ebf5f3
[2019-05-13 23:44:51,629] [DEBUG] -- <Response [200]>

[2019-05-13 23:44:51,629] [DEBUG] -- assign_sla: Assigning the MSSQL Instance '<host_redacted>' to the 'Gold' SLA Domain.
[2019-05-13 23:44:51,629] [DEBUG] -- Creating the authorization header using the provided username and password.
[2019-05-13 23:44:51,629] [DEBUG] -- PATCH https://<ip_redacted>/api/v1/mssql/instance/MssqlInstance:::1b906a15-8197-4408-9ae3-109068ebf5f3
[2019-05-13 23:44:51,630] [DEBUG] -- Config: {"configuredSlaDomainId": "8e6f6618-6af0-4833-a94d-6b32f00ca9b0", "logRetentionHours": 12, "logBackupFrequencyInSeconds": 600, "copyOnly": false}
[2019-05-13 23:44:57,236] [DEBUG] -- <Response [200]>

>>> print assignsla
[{u'rootProperties': {u'rootType': u'Host', u'rootId': u'Host:::32c0637b-6398-4f66-816c-044b2073b3cf', u'rootName': u'<host_redacted>'}, u'logBackupFrequencyInSeconds': 600, u'name': u'MSSQLSERVER', u'primaryClusterId': u'603109f2-eb30-4da8-9389-911d66abb524', u'configuredSlaDomainName': u'Gold', u'unprotectableReasons': [], u'id': u'MssqlInstance:::1b906a15-8197-4408-9ae3-109068ebf5f3', u'version': u'11.0.6020.0', u'configuredSlaDomainId': u'8e6f6618-6af0-4833-a94d-6b32f00ca9b0', u'internalTimestamp': 1557787493431000, u'protectionDate': u'2019-05-13T22:44:53.431Z', u'logRetentionHours': 12, u'copyOnly': False}]
```

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rubrik_cdm/data_management.py [hunk 1]
```diff
@@ -377,6 +377,58 @@ def assign_sla(self, object_name, sla_name, object_type, timeout=30):
...
+
+                        config['configuredSlaDomainId'] = sla_id
+
+                        patch_resp = self.patch("v1", "/mssql/instance/{}".format(mssql_id), config, timeout)
+                        db_sla_lst.append(patch_resp)
+            else:
+                raise InvalidParameterException(
+                    "Host ID not found for instance '{}'").format(object_name)
+
+            return db_sla_lst
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: How come we return the list here and not just the response of the PATCH call?

Comment (drew-russell) [reply rubrik_cdm/data_management.py]: We can also just return that right away after

````py
db_sla_lst.append(patch_resp)
 return db_sla_lst
```

Comment (Draper1) [reply rubrik_cdm/data_management.py]: The code loops through all instances on the MSSQL Host and patches each instance (This is how it functions in our UI).

Comment (drew-russell) [reply rubrik_cdm/data_management.py]: Ahhh, got it.

**Pull request merged**